weave-js/src/components/PagePanelComponents/Home/Browse3/context.tsx

```diff
@@ -624,6 +625,46 @@ export const WeaveflowPeekContext = createContext<{
   isPeeking: false,
 });
 
+export const WeaveMainTableContext = createContext<{
+  mainTableRef: React.MutableRefObject<GridApiPro> | undefined;
+  isRefSet: boolean;
+  setIsRefSet: (isSet: boolean) => void;
+  setMainTableRef: (
+    ref: React.MutableRefObject<GridApiPro> | undefined
+  ) => void;
+}>({
+  isRefSet: false,
```

Comment:
It is not immediately obvious to me why we have these 4 items. Can you comment?

Reply:
should we show the button, and setting the ref happen lower in the component structure. So we expose the ability to set the ref and set shouldShowTheExportButton.